### PR TITLE
[keccak-hash] Add no_std support, migrate to 2018 edition

### PR DESCRIPTION
**Pull request by koushiro**
Signed-off-by: koushiro <koushiro.cqx@gmail.com>

# What have I changed

- Add no_std support;

- Migrate keccak-hash to 2018 edition;

related issue: #143

### CODE REVIEW

File: keccak-hash/Cargo.toml [hunk 1]
```diff
@@ -6,10 +6,15 @@ authors = ["Parity Technologies <admin@parity.io>"]
...
 license = "GPL-3.0"
+edition = "2018"
 
 [dependencies]
 tiny-keccak = "1.4"
-primitive-types = { path = "../primitive-types", version = "0.4" }
 
 [dev-dependencies]
 tempdir = "0.3"
+rustc-hex = { version = "2.0", default-features = false }
```
Comment (dvdplm) [inline keccak-hash/Cargo.toml]: Please revert this in favour of `hex_literal`. We are trying to slowly phase out `rustc-hex` and it should not be used by new code.

Comment (koushiro) [reply keccak-hash/Cargo.toml]: Okay

**Comment by koushiro** [comment]
Updated, PTAL again @dvdplm

**Comment by ordian** [comment]
@dvdplm could you take a look again?

File: keccak-hash/src/lib.rs [hunk 2]
```diff
@@ -14,15 +14,17 @@
...
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
-extern crate primitive_types;
-extern crate tiny_keccak;
+#![cfg_attr(not(feature = "std"), no_std)]
 
-use std::io;
-use std::slice;
+#[cfg(not(feature = "std"))]
+extern crate alloc;
```
Comment (dvdplm) [inline keccak-hash/src/lib.rs]: Doesn't `use alloc;` work here?

**Comment by ordian** [comment]
@debris can I spend a review coin here? :)

**Pull request merged**